src/snowflake/connector/cursor.py

```diff
@@ -264,6 +265,7 @@ def __init__(
         self._inner_cursor: SnowflakeCursor | None = None
         self._prefetch_hook = None
         self._rownumber: int | None = None
+        self._last_response_dict: dict | None = None
```

Comment:
Apologies for thinking out loud again and any churn that might cause.

Generally, and especially with type hints, we don't include the type of the data structure the variable is bound to in the name of the variable.  Now that I'm looking at this again, what I think is interesting is that the property `_last_response_json()` actually returns a string.  In this case though you just need to make sure that the internal variable and the property name don't collide.

What do you think about calling this variable `self._cached_last_result` and the property `_last_result()`?

The other question I have is whether you want to access the last result as a string or as a dictionary.  In the test, you deserialize the string to check its contents.  Why not have the property return the cached last result dictionary directly, rather than serializing it to a string?

Reply:
yeah, I do think `self._cached_last_result` and `_last_result` makes sense. Also, I have updated the return type to be a dictionary instead of JSON string.